training/model_management/components/import_model/spec.yaml

```diff
@@ -0,0 +1,135 @@
+$schema: https://azuremlschemas.azureedge.net/latest/pipelineComponent.schema.json
+type: pipeline
+
+name: import_model
+display_name: Import model
+description: Import a model into a workspace or a registry
+version: 0.0.1
+
+# Pipeline inputs
+inputs:
+  # pipeline specific compute
+  compute:
+    type: string
+    description: Compute to run pipeline job
+
+  ## Inputs for download model
+  model_source:
+    type: string
+    description: Storage containers from where model will be sourced from.
+    default: Huggingface
+    enum:
+      - AzureBlob
+      - GIT
+      - Huggingface
+
+  model_id:
+    type: string
+    description: A valid model id for the model source selected. For example you can specify `bert-base-uncased` for importing HuggingFace bert base uncased model. Please specify the complete URL if **GIT** or **AzureBlob** is selected in `model_source`
+
+  ## Inputs for the MlFLow conversion
+  license_file_path:
+    type: uri_file
+    description: Path to the license file
+    optional: true
+
+  ## Inputs for model local validation
+  test_data_path:
+    type: uri_file
+    optional: true
+    description: Test dataset for model inferencing
+
+  test_data_column_rename_map:
+    type: string
+    optional: true
+    description: |
+      Provide mapping of dataset column names that should be renamed before inferencing.
+      eg: col1:ren1; col2:ren2; col3:ren3
+
+  ## Inputs for Model registration
+  custom_model_name:
+    type: string
+    optional: true
+    description: Model name to use in the registration. If name already exists, the version will be auto incremented
+
+  model_version:
+    type: string
+    optional: true
+    description: Model version in workspace/registry. If the same model name and version exists, the version will be auto incremented
+
+  model_description:
+    type: string
+    optional: true
+    description: Description of the model that will be shown in AzureML registry or workspace
+
+  registry_name:
+    type: string
+    optional: true
+    description: Name of the AzureML asset registry where the model will be registered. Model will be registered in a workspace if this is unspecified
+
+  model_metadata:
+    type: uri_file
+    optional: true
+    description: A JSON or a YAML file that contains model metadata confirming to Model V2 [contract](https://azuremlschemas.azureedge.net/latest/model.schema.json)
+
+# Pipeline outputs
+outputs:
+  model_registration_details:
+    description: Output file which captures transformations applied above and registration details in JSON file.
+    type: uri_file
```

Comment:
Can we output an MLFlow model?

Reply:
What additional information does this uri file have?

Reply:
Can we output an MLFlow model?
=> Yes, we can. Can you help to share a use case? Most registration there might be interest in deployment for which model registration details would be sufficient. For finetuning and evaluation purposes, registered model can be used as an input. But it cannot be part of same pipeline job.

    What additional information does this uri file have?
=> model information is captured here, which would include following details:
```
 model_info = {
        "id": registered_model.id,
        "name": registered_model.name,
        "version": registered_model.version,
        "path": registered_model.path,
        "flavors": registered_model.flavors,
        "type": registered_model.type,
        "properties": registered_model.properties,
        "tags": registered_model.tags,
        "description": registered_model.description,
    }
```

Reply:
Got it. Let us discuss internally once publish this version.